### DISCUSSION TITLE
system.random()

### DISCUSSION THREAD

**Discussion by kroggen** [aergoio/aergo#275] [General]
This is a place to discuss the `system.random()` function

**Comment by kroggen** [comment]
The `system.random()` is not random when used on queries run at the same block height

If many queries are sent to the node, it will generate the same value for all the queries

Is this a desired behavior?

The queries that are run against the next block will generate a different number

I suspect that there is no need for deterministic behavior on queries

**Reply by kroggen** [reply]
Related code [here](https://github.com/aergoio/aergo/blob/aaa3e4bdfb65ee3d67cd9610a428e04e02b8167c/contract/vm.go#L903)

It could be a true random source here, different for each query

Or a constant, and so all queries would return the same value regardless of which block height they are run

**Comment by gosunuts** [comment]
This is a about oracle problem in blockchain. https://en.wikipedia.org/wiki/Blockchain_oracle
All nodes require the same operations for deterministic, so many blockchain use block height to make time and random variables.
Even if it is not the block height, it needs to be a standard seed that all nodes can agree.

**Reply by kroggen** [reply]
This is for calls, or transactions processed by all nodes

But queries are processed by a single node, no?

**Reply by kroggen** [reply]
I mean, the code for calls is OK

The question I posted above is about queries

**Reply by gosunuts** [reply]
i understand. well, if query doesn`t consume gas, it is not problem.

**Reply by gosunuts** [reply]
But, is it worth it? It is literally 'random', and blockchain can`t guarantee about these random value. what is different to make random outside of nodes?

**Reply by gosunuts** [reply]
In my opinion, it sounds weird that random is generated from blockchain.
